### DISCUSSION TITLE
Email verification

### DISCUSSION THREAD

**Discussion by hadi-codes** [supabase/supabase#1601] [Questions] [answered]
Hello,
I was wondering if there is a way to check if an email has been verified after signing up ?.

**Comment by wadclapp** [comment]
One way of doing it, maybe
```sql
auth.uid() in ( select id from auth.users WHERE email_confirmed_at is not null )
```
That's assuming `email_confirmed_at` gets reset to NULL if the user changes email. Haven't tested, can't find documentation on it, and I expect there is (or will be) a better way

**Comment by w3b6x9** [comment]
@bigblackclock i would do it the way @wadclapp mentioned and create a security definer function to see if an email is confirmed and call it via `rpc`.

There's an issue regarding email update here so it's unclear how to do it via the client and `auth.user()` (see: https://github.com/supabase/gotrue-js/issues/57).

I would also check to see if `email_change` is empty b/c that's used during the email change process.

So I would probably validate email is confirmed by checking that `email_confirmed_at` is not null and `email_change` is an empty string. However, I'm not 100% about these two checks to confirm email so proceed with caution!

**Reply by wadclapp** [reply]
In Firebase it's simply an `emailVerified` property on the auth object. Would be good to get something similar

**Comment by J0** [comment, marked as answer]
Hey all,

 Thanks for the queries! We now expose an `email_confirmed_at` property on the user which one can make use of to see if an email has been verified.

Thanks

**Reply by nuurcodes** [reply]
If 'Confirm Email' is disabled in Email Provider settings then email_confirmed_at is set at account creation.

Sometimes we want the user to be able to access the app with an unconfirmed email and ask them to confirm later. 
I also saw there is a confirmed_at in the auth users table. Maybe setting this is enough to let the user log in with an unconfirmed email?
